src/lib/twizzler-runtime-api/src/lib.rs

```diff
@@ -50,8 +51,68 @@ use core::{
 #[cfg(feature = "rt0")]
 pub mod rt0;
 
-/// Core object ID type in Twizzler.
-pub type ObjID = u128;
+#[repr(transparent)]
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
+/// An object ID, represented as a transparent wrapper type. Any value where the upper 64 bits are
+/// zero is invalid.
```

Comment:
I forget, is the rationale here that object ids with the upper 64 bits being zero are reserved by the system?

Reply:
Yep!